src/app/auth/register/register.component.ts

```diff
@@ -1,22 +1,104 @@
-import { Component, OnInit } from '@angular/core';
-import { FormControl, FormGroup } from '@angular/forms';
+import { Component, OnDestroy, OnInit, OnChanges, DoCheck } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { sameAsEmailValidator } from "./verify/validate/same-as-email.directive";
+import { samePassValidator } from "./verify/validate/pass-match.directive";
+import { HttpResponse, HttpClient } from '@angular/common/http';
 
 @Component({
   selector: 'pluma-online-register',
   templateUrl: './register.component.html',
   styleUrls: ['./register.component.scss']
 })
-export class RegisterComponent implements OnInit {
-  public registerForm: FormGroup = new FormGroup({
-    email: new FormControl(''),
-    password: new FormControl('')
-  });
-
+export class RegisterComponent implements OnInit, OnDestroy, DoCheck {
+  public registerForm: FormGroup;
   public isSubmitted: boolean = false;
+  public oldPass: string = '';
+  public oldEmail: string = '';
+  public oldDName: string = '';
+  public uniqueE: boolean = true;
```

Comment:
Bad naming sense from my part. Should be "uniqueEmailFlag". It gets flagged false if backend response returns with E_ERR or B_ERR. This is so that user knows why submission did not pass. Do you want me to do this a different way?

Reply:
Use HTTP statuses and messages from them.